### ISSUE TITLE
AsmSelect should mark items for deletion rather than remove them immediately

### ISSUE THREAD

**Issue by Toutouwai** [processwire/processwire-requests#55]
### Short description of the enhancement

It would be more consistent with other multi-value inputfields (File, Image, Repeater, PageTable) if clicking the "trash" icon for an item in an AsmSelect marked it for deletion rather than removed it immediately from the inputfield.

### Why would the enhancement be useful to users?

1. It would be more consistent with other multi-value inputfields and therefore meet expectations of how the field will behave.

2. It would give the opportunity to restore an item before page save if you change your mind.

3. It would serve as a visual reminder of how the field will change at the time the page is saved. For instance, you are doing a number of edits to a page and before saving you want a reminder that the AsmSelect was modified and what the deletions will be. In that regard, it might also be a nice enhancement if newly added items were marked visually somehow.

**Comment by szabeszg** [comment]
While it is generally a good idea, I would not get rid of the current behavior either. There are times when it just good enough the way it is, and being removed immediately provides a fast workflow.
It should be optional which method we want to use, and it should be set at field level.

**Comment by ryancramerdesign** [comment]
asmSelect already supports this option, via the "hideDeleted" property,
which can be set to "false". I've set it to use this in a few places where
it seemed worthwhile, like when editing a Template, and the "Fields"
asmSelect, in FormBuilder, and elsewhere. Are you specifically looking for
this as a config option like for Page fields?

**Comment by Toutouwai** [comment]
Yes, a field config option would be good. Personally I think it should be the default option for the reasons above. Not sure I understand @szabeszg's comment about fast workflow as neither would be any faster than the other as the deletions are not applied until page save in either case.

**Comment by szabeszg** [comment]
Sure, the word "fast" was a bad one to choose, since there is really just an edge use case to which my remark applies and that is when we have lots of items and we also want to re-arrange them, in which case it might be a bit hard to organize the items with loads of no longer needed ones still in the list. But probably this is what happens rarely, so turning this new proposal into the default option is a good idea, I think.
